packages/repository-json-schema/src/build-schema.ts

```diff
@@ -165,12 +165,12 @@ export function modelToJsonSchema(
     return {};
   }
 
-  const title = meta.title || ctor.name;
+  const title = meta.title || meta.name || ctor.name;
 
-  if (options.visited[title]) return options.visited[title];
+  if (options.visited.has(ctor)) return options.visited.get(ctor)!;
```

Comment:
Please note we are going to create multiple `visited` entries for a single constructor in the very near feature:
 - schema for the model as-is
 - schema for the model with `id` and `_rev` properties excluded
 - schema for the model and navigational properties for relations

See the discussion in the pull request that introduced `options.visited` for more context.

Please revert this change and keep using `title` as the key.